src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,57 @@
+@import "../utils/_variables";
+@import "../utils/normalize";
+
+@mixin square ($side) {
+  height: $side;
+  width: $side;
+}
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  width: $calendar-box-size;
+  gap: $box-gap;
+  padding: 20px 0;
+  margin: 0 auto;
+  align-items: center;
+
+  @for $i from 1 through 31 {
+    & :nth-child(#{$i})::before {
+      content: "#{$i}";
+    }
+  }
+  
+  @each $day, $dayNum in $days-of-week {
+    &--start-day--#{$day} :first-child {
+      margin-left: $dayNum * ($calendar-dey-box + $box-gap);
+    }
+  }
+  
+  @for $i from 28 through 31 {
+    &--month-length--#{$i} :nth-child(n + #{$i + 1}) {
+      display: none;
+    }
+  }
+
+  &__day {
+    position: relative;
+    @include square($calendar-dey-box);
+
+    background-color: #eee;
+    border: 1px solid black;
+    transition-duration: 0.5s;
```

Comment:
don't use it like that, because it's the same as transition: all 0.5s
It's too hard for the browser
Check this
![image](https://user-images.githubusercontent.com/91411692/222707020-db2cf7b4-14f0-453e-b294-fb9dfd47ffad.png)

also, create a variable for transition Duration
```suggestion
    transition: transform $transitionDuration;
```